### ISSUE TITLE
Wrong signature in received PDU

### ISSUE THREAD

**Issue by tguillem** [sahlberg/libsmb2#176]
When connecting to a samba Version 4.12.5-Debian with the following configuration:

```
min protocol = SMB2
max protocol = SMB2
server signing = mandatory
```

smb2_service() fails with the following error: "Wrong signature in received PDU".

The error come from smb2_read_data() in lib/socket.c.

Note that the PDU signature check is OK from session_setup_cb() in lib/libsmb2.c

Removing the PDU signature check from lib/socket.c (and letting it in lib/libsmb2.c) is fixing my issue.

What is the proper solution?

**Comment by delins** [comment]
@tguillem A similar issue was fixed in #183. Could you check if the latest commit fixes this issue as well?

**Comment by tguillem** [comment]
@delins I confirm that the lasted commit fixes this issue. Thanks a lot !
